commit 3b4fc3bd0f87bf97b02a1a5a2cc1e6b3c80c80ca (refs/changes/39/2639/2)
Author: Jason King <jason.brian.king@gmail.com>
Date:   2017-09-22T18:26:13+00:00 (2 years, 1 month ago)
    
    OS-6362 coreadm(1M) allows one to specify bad global patterns
    Reviewed by: Robert Mustacchi <rm@joyent.com>

diff --git a/usr/src/cmd/coreadm/coreadm.c b/usr/src/cmd/coreadm/coreadm.c
index 9ce0f27651..f47f595ca3 100644
--- a/usr/src/cmd/coreadm/coreadm.c
+++ b/usr/src/cmd/coreadm/coreadm.c
@@ -21,6 +21,8 @@
 /*
  * Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
+ *
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 #include <stdio.h>
@@ -243,6 +245,12 @@ main(int argc, char **argv)
 		    command);
 		usage();
 	}
+	if (glob_pattern != NULL && glob_pattern[0] != '/') {
+		(void) fprintf(stderr, gettext(
+		    "%s: The -g option must specify an absolute path\n"),
+		    command);
+		usage();
+	}
 	if ((proc_pattern != NULL || proc_content != CC_CONTENT_INVALID) &&
 	    npids == 0) {
 		(void) sprintf(curpid, "%u", (uint_t)getppid());
